system/lib/wasmfs/file.cpp

```diff
@@ -83,6 +85,15 @@ std::shared_ptr<Directory> getDir(std::vector<std::string>::iterator begin,
 #endif
     curr = directory->locked().getEntry(*it);
 
+    // For the rename syscall one must check if the source dir is an ancestor of
+    // the destination dir.
+    if (ancestor) {
+      if (curr == ancestor) {
+        err = -EINVAL;
+        return nullptr;
+      }
+    }
```

Comment:
It looks like "ancestor" is something we stop and error at. How about calling it `stopAt` or `errorOn`?

Reply:
or `forbiddenAncestor`?

Reply:
Ah, nice, `forbiddenAncestor` is even clearer!